homeassistant/components/ai_speaker/sensor.py

```diff
@@ -0,0 +1,96 @@
+"""Support for AI-Speaker sensors."""
+from datetime import timedelta
+import logging
+
+from homeassistant.const import DATA_RATE_KILOBITS_PER_SECOND
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = timedelta(minutes=60)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Perform the setup for AI-Speaker status sensor."""
+    _LOGGER.debug("AI-Speaker sensor, async_setup_entry")
+    ais_gate_instance = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([AisSensor(ais_gate_instance)], True)
+
+
+class AisSensor(Entity):
+    """AiSpeakerSensor representation."""
+
+    def __init__(self, ais_gate_instance):
+        """Sensor initialization."""
+        self._ais_gate = ais_gate_instance
+        self._ais_id = None
+        self._ais_product = None
+        self._ais_manufacturer = None
+        self._ais_model = None
+        self._ais_os_version = None
+        self._ais_api_level = None
+        self._ais_ip_address = None
+        self._ais_hostname = None
+        self._ais_gate_network_speed = None
+
+    @property
+    def device_info(self):
+        """Device info."""
+        return {
+            "identifiers": {(DOMAIN, self._ais_id)},
+            "name": f"AI-Speaker {self._ais_product}",
+            "manufacturer": self._ais_manufacturer,
+            "model": self._ais_model,
+            "sw_version": self._ais_os_version,
+            "via_device": None,
+        }
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique, friendly identifier for this entity."""
+        return self._ais_id
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"AI-Speaker {self._ais_product} status"
+
+    @property
+    def state(self):
+        """Return the status of the sensor."""
+        return self._ais_gate_network_speed
```

Comment:
This doesn't match the name. The name is "status"

Reply:
sorry I don't get this. Could you please explain?

Reply:
The network speed value of a device isn't a "status". A status is a state like "Ready", "Listening" etc. It's not a numeric measurement? It seems that either the state or the name is incorrect.

Reply:
Thanks, I finally understand this (I hope) ;) 
I'm changing a sensor name to:
"AI-Speaker {self._ais_product} network connection speed."

Reply:
What would be the use case of having a network connection speed sensor? Maybe that could be part of system health, but I don't see how that would be relevant as an entity?

Reply:
Hi @frenck  :wave: 
The speaker is a network streamer. The network connection speed can be useful info for the user. Of course if there is better way to present device status with attributes then I'm open for suggestions?
BTW
The main goal for this integration is to provide media player with media browser, then voice assistant, then camera etc. But this is my first pull request to HA, I need to start from one small functionality.

Reply:
> The main goal for this integration is to provide media player with media browser, then voice assistant, then camera etc.

☝️  That sounds useful. This sensor being added right now, isn't an entity worthy IMHO and should not be added.

> present device status with attributes then I'm open for suggestions?

I gave that suggestion in my previous response? Sounds more like a system health thing to check when one is having a connection problem or something.

Reply:
> That sounds useful. 

yes, but this functionality was removed from this pull because first pull can have only one small thing. 
Now we have configuration flow and sensor with device network connection status. This is not so useful for end user like browsing and playing media but this is a start.

> I gave that suggestion in my previous response? Sounds more like a system health thing (shown in the Configuration panel -> Info) to check when one is having a connection problem or something.

yes you did, but I'm not sure I get it ;) Could you please show me the code - how this kind of functionality is developed by other integrations? 
We are not doing anything wrong here, the pull to get connection speed info/status once per h. Having this info as sensor is like a easy way to have this info in app, user can do some automation based on this, present this on chart etc. 
IMHO I would like to have this info. I'm open to make this integration as useful as possible. I did all according to instructions from HA. Please guide me to make this better.

---------------------------------
Of course it is your project. So decision is yours. 
If you want to drop this sensor and implement media player as a first integration I can switch to media player.
If you want to close this pull, feel free too, I will not cry ;) I learned a lot - thanks!
My suggestion is to mark this integration as BETA (like Tasmota etc.) and merge this first pull request is the implementation is OK. Then I will deliver the more useful functionality in next pull requests.

Reply:
If I can add my 5 cents. I would like to have a network speed from AI-Speaker device as a sensor, I don't see anything strange here.

I really like that you @frenck are checking what can be delivered to core - thank you for this! This project should be (and I think it will be) a pattern for many big Open Source software projects.

But if all the technical aspects and requirements are met, then the personal opinion what will be useful is not so significant. Home Assistant is Open Source platform, tested, used and improved by community / nobody is forcing to install this integration if for someone it is not useful. 

We should appreciate the work done by @asystentka-jolka, who want to deliver and maintain this integration, otherwise this will be very demotivational. IMHO.

Reply:
> But if all the technical aspects and requirements are met, then the personal opinion what will be useful is not so significant. Home Assistant is Open Source platform, tested, used and improved by community / nobody is forcing to install this integration if for someone it is not useful.

That comment makes 0 sense. This is not a personal opinion, this is based on our regular architectural design. I don't use this, thus I'm not personally influenced.

> yes, but this functionality was removed from this pull because first pull can have only one small thing.

That is actually not completely correct. It is recommended to keep the first PR small, sure.

See: <https://developers.home-assistant.io/docs/creating_component_code_review#5-make-your-pull-request-as-small-as-possible>

It states: Limit to a single platform

There is no limit on type of platform, single sensor or not being able to do something like media player. Just to be clear on that.

> yes you did, but I'm not sure I get it ;) Could you please show me the code - how this kind of functionality is developed by other integrations?

Some PRs that added System Health:

<https://github.com/home-assistant/core/pulls?q=is%3Apr+sort%3Aupdated-desc+system+health+is%3Aclosed>

Reply:
hi @frenck thanks for the link :+1: 
I will check this later it detail, looks interesting... I never saw any info about system_health  in dev documentation (don't had idea that this exists), that is why I asked.

Just to be sure, should I start from the system_health in this pull or swith to media_player and deliver the system_health later on... what is the propoer way to go on with this pull request?

--------------
@others could you please relax :)

Reply:
> what is the propoer way to go on with this pull request?

Add one or more entities from a single platform, that provides some actual use for the end-user the first time they use this integration.

Generally, consider this question: What is the entity that everybody would instantly use and most use when they install this integration? <- The answer to that question is generally a good start.

Reply:
> Generally, consider this question: What is the entity that everybody would instantly use and most use when they install this integration? <- The answer to that question is generally a good start.

Thanks, I will go with media_player (the minimal implementation at first). Hold on... ;)